core/arch/arm/kernel/thread.c

```diff
@@ -294,7 +294,7 @@ static void thread_lazy_save_ns_vfp(void)
 	struct thread_ctx *thr = threads + thread_get_id();
 
 	thr->vfp_state.ns_saved = false;
-#if defined(ARM64) && defined(CFG_WITH_ARM_TRUSTED_FW)
+#if defined(ARM64) || defined(CFG_WITH_ARM_TRUSTED_FW)
```

Comment:
If I understand correctly, `defined(ARM64)` is not needed, the TEE core being 32 or 64-bit is unrelated to the need to force save or not -> remove

Reply:
Correct, I'll remove the `defined(ARM64)`